Demos/Device/ClassDriver/CCID/Descriptors.c

```diff
@@ -0,0 +1,219 @@
+ *  device characteristics, including the supported USB version, control endpoint size and the
+ *  number of device configurations. The descriptor is read out by the USB host when the enumeration
+ *  process begins.
+ */
+const USB_Descriptor_Device_t PROGMEM DeviceDescriptor =
+{
+	//It  is  a  standard  device  descriptor  as  per  Chapter  9,  “USB  Device  Framework,”  in  the   Universal Serial Bus Specification.
+	//It does not contain class-specific information.  
+	.Header                 = {.Size = sizeof(USB_Descriptor_Device_t), .Type = DTYPE_Device},
+
+	.USBSpecification       = VERSION_BCD(1,1,0),
+	.Class                  = USB_CSCP_NoDeviceClass,
+	.SubClass               = USB_CSCP_NoDeviceSubclass,
+	.Protocol               = USB_CSCP_NoDeviceProtocol,
+
+	.Endpoint0Size          = FIXED_CONTROL_ENDPOINT_SIZE,
+
+	.VendorID               = 0x03EB,
+	.ProductID              = 0x206E,
+	.ReleaseNumber          = VERSION_BCD(0,0,1),
+
+	.ManufacturerStrIndex   = STRING_ID_Manufacturer,
+	.ProductStrIndex        = STRING_ID_Product,
+	.SerialNumStrIndex      = NO_DESCRIPTOR,
+
+	.NumberOfConfigurations = FIXED_NUM_CONFIGURATIONS
+};
+
+/** Configuration descriptor structure. This descriptor, located in FLASH memory, describes the usage
+ *  of the device in one of its supported configurations, including information about any device interfaces
+ *  and endpoints. The descriptor is read out by the USB host during the enumeration process when selecting
+ *  a configuration so that the host may correctly communicate with the USB device.
+ */
+const USB_Descriptor_Configuration_t PROGMEM ConfigurationDescriptor =
+{
+	.Config =
+		{
+			.Header                   = {.Size = sizeof(USB_Descriptor_Configuration_Header_t), .Type = DTYPE_Configuration},
+
+			.TotalConfigurationSize   = sizeof(USB_Descriptor_Configuration_t),
+			.TotalInterfaces          = 1,
+
+			.ConfigurationNumber      = 1,
+			.ConfigurationStrIndex    = NO_DESCRIPTOR,
+
+			.ConfigAttributes         = (USB_CONFIG_ATTR_RESERVED | USB_CONFIG_ATTR_SELFPOWERED),
+
+			.MaxPowerConsumption      = USB_CONFIG_POWER_MA(100)
+		},
+		.CCID_Interface =
+		{
+			.Header                 = {.Size = sizeof(USB_Descriptor_Interface_t), .Type = DTYPE_Interface},
+
+			.InterfaceNumber        = INTERFACE_ID_CCID,
+			.AlternateSetting       = 0x00,
+
+			.TotalEndpoints         = 2,
+
+			.Class                  = CCID_CSCP_CCIDClass, 
+			.SubClass               = CCID_CSCP_NoSpecificSubclass,
+			.Protocol               = CCID_CSCP_NoSpecificProtocol,
+
+			.InterfaceStrIndex      = NO_DESCRIPTOR
+		},
+		.CCID_SmartCard =
+		{
+			.Header                 = {.Size = sizeof(USB_CCID_Descriptor_t), .Type = DTYPE_FunctionalDescriptor},
+			.CCID  					= CCID_CURRENT_SPEC_RELEASE_NUMBER,
+			.MaxSlotIndex			= 0x00,
+			.VoltageSupport			= CCID_VOLTAGESUPPORT_5V,
+			.Protocols				= CCID_PROTOCOLS_T1,
+			.DefaultClock			= 16000, //16MHz
```

Comment:
Might be worth a `CCID_DESCRIPTOR_CLOCK(hz)` or similar macro in the library, which does the /1000 scaling for the user.